src/Windows/Avalonia.Win32/WindowImpl.CustomCaptionProc.cs

```diff
@@ -11,70 +10,78 @@ namespace Avalonia.Win32
 
-            RECT border_thickness = new RECT();
+            var borderThickness = new RECT();
             if (GetStyle().HasFlag(WindowStyles.WS_THICKFRAME))
             {
-                AdjustWindowRectEx(ref border_thickness, (uint)(GetStyle()), false, 0);
-                border_thickness.left *= -1;
-                border_thickness.top *= -1;
+                AdjustWindowRectEx(ref borderThickness, (uint)(GetStyle()), false, 0);
+                borderThickness.left *= -1;
+                borderThickness.top *= -1;
             }
             else if (GetStyle().HasFlag(WindowStyles.WS_BORDER))
             {
-                border_thickness = new RECT { bottom = 1, left = 1, right = 1, top = 1 };
+                borderThickness = new RECT { bottom = 1, left = 1, right = 1, top = 1 };
             }
 
             if (_extendTitleBarHint >= 0)
             {
-                border_thickness.top = (int)(_extendedMargins.Top * RenderScaling);
+                borderThickness.top = (int)(_extendedMargins.Top * RenderScaling);
             }
 
             // Determine if the hit test is for resizing. Default middle (1,1).
             ushort uRow = 1;
             ushort uCol = 1;
-            bool fOnResizeBorder = false;
+            bool onResizeBorder = false;
 
-            // Determine if the point is at the top or bottom of the window.
-            if (ptMouse.Y >= rcWindow.top && ptMouse.Y < rcWindow.top + border_thickness.top)
+            // Determine if the point is at the left or right of the window.
+            if (ptMouse.X >= rcWindow.left && ptMouse.X < rcWindow.left + borderThickness.left)
             {
-                fOnResizeBorder = (ptMouse.Y < (rcWindow.top - rcFrame.top));
-                uRow = 0;
+                uCol = 0; // left side
             }
-            else if (ptMouse.Y < rcWindow.bottom && ptMouse.Y >= rcWindow.bottom - border_thickness.bottom)
+            else if (ptMouse.X < rcWindow.right && ptMouse.X >= rcWindow.right - borderThickness.right)
             {
-                uRow = 2;
+                uCol = 2; // right side
             }
 
-            // Determine if the point is at the left or right of the window.
-            if (ptMouse.X >= rcWindow.left && ptMouse.X < rcWindow.left + border_thickness.left)
+            // Determine if the point is at the top or bottom of the window.
+            if (ptMouse.Y >= rcWindow.top && ptMouse.Y < rcWindow.top + borderThickness.top)
             {
-                uCol = 0; // left side
+                onResizeBorder = (ptMouse.Y < (rcWindow.top - rcFrame.top));
+
+                // Two cases where we have a valid row 0 hit test:
+                // - window resize border (top resize border hit)
+                // - area below resize border that is actual titlebar (caption hit).
+                if (onResizeBorder || uCol == 1)
```

Comment:
This is the main part of the change - we can't assume that being in the titlebar area is row 0 since that does not apply to corners.